### PR TITLE
fix: remove `libgcc_s_seh-1.dll` dependency

### PR DESCRIPTION
**Pull request by Seb-MCaw**
It seems 134d1157 added a dependency on `libgcc_s_seh-1.dll` (via the new `den` crate) on Windows, without which all commands instantly (and silently) fail. This DLL is packaged along with GNAT but, since `alr` should be able to auto-install the toolchain on a fresh system, this behaviour seems like a bug.

This PR adds the `-static` linker switch, removing the external dependency on `libgcc_s_seh-1.dll`.

##### PR creation checklist
- [x] A test is included, if required by the changes.
- [ ] `doc/user-changes.md` has been updated, if there are user-visible changes.
- [ ] `doc/catalog-format-spec.md` has been updated, if applicable.
- [ ] `BREAKING.md` has been updated for major changes in `alr`, minor/major in catalog format.

### CODE REVIEW

**Review by mosteo — APPROVED** [review]

**Comment by mosteo** [comment]
Yes, indeed, and not long ago I added a check in our GH actions to check that Alire works even after the toolchain that built it is removed. I don't really understand why it didn't catch this problem though.

**Comment by Seb-MCaw** [comment]
> I don't really understand why it didn't catch this problem though.

I believe the GitHub Windows runners are using Git Bash, which puts a MinGW distribution on `PATH` by default, so it is available at `/mingw64/bin/libgcc_s_seh-1.dll` even without GNAT installed.

File: testsuite/tests/misc/empty-path/test.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+from drivers.alr import run_alr
+
+
+# Check `alr --version` succeeds and prints something.
+p = run_alr("--version")
+assert len(p.out) > 0
+
+# Completely clear `PATH`.
+#
+# Setting it to an empty string causes a buffer overflow on some platforms
```
Comment (Seb-MCaw) [inline testsuite/tests/misc/empty-path/test.py:17]: This probably technically qualifies as a bug, but anybody with an empty `PATH` probably has bigger problems than `alr` failing, so I presume we can safely ignore it. Let me know if you want me to open an issue, though.

Comment (mosteo) [reply testsuite/tests/misc/empty-path/test.py:17]: You mean `alr` is the one having the buffer overflow?

Comment (Seb-MCaw) [reply testsuite/tests/misc/empty-path/test.py:17]: [Yes, the subsequent `alr --version` command fails](https://github.com/alire-project/alire/actions/runs/13291811450/job/37114259324#step:4:1515).

Comment (mosteo) [reply testsuite/tests/misc/empty-path/test.py:17]: Ouch. Then yes, even if it's an impractical corner case, I think we should track it.
